src/main/reactjs/src/components/clerkTranslator/overview/ClerkTranslatorField.tsx

```diff
@@ -0,0 +1,64 @@
+import { TextFieldProps } from '@mui/material';
+import { ChangeEvent } from 'react';
+
+import { CustomTextField } from 'components/elements/CustomTextField';
+import { useAppTranslation } from 'configs/i18n';
+import { TextFieldTypes } from 'enums/app';
+import {
+  ClerkTranslator,
+  ClerkTranslatorDetailsAttribute,
+} from 'interfaces/clerkTranslator';
+import { CustomTextFieldProps } from 'interfaces/customTextField';
+import { Utils } from 'utils';
+
+const getFieldType = (field: ClerkTranslatorDetailsAttribute) => {
+  switch (field) {
+    case 'phoneNumber':
```

Comment:
Tässä voisi käyttää jotain enumiä tai tms. Field.jotain

Reply:
Tässä siis ideana että interfacen puolella on määritetty `export type ClerkTranslatorDetailsAttribute = keyof ClerkTranslatorBasicDetails;`. En oikein tiedä miten tuon enum-idean saisi toimimaan ilman (mielestäni turhaa) toistoa.

Reply:
jep, sitä ei varmaan saa ilman toistoa, mutta toisaalta parempi kuin noi stringit. Varsinkin kun niitä aletaan käyttää muualla niin voi aiheuttaa ongelmia. Tässä voisi miettiä myös olisiko silti keyof ClerkTranslatorBasicDetails kuvaavampi kuin ClerkTranslatorDetailsAttribute. Noita termejä alkaa olla meillä liikkaa :D

Reply:
Joo, ehkä se `keyof ClerkTranslator...` on pienempi paha.

Reply:
En silti oo kovin innoissani ajatuksesta tehdä erillistä enumia noille eri kentille. Se olisi taas yksi uusi ylläpidettävä tietorakenne.

Reply:
niin ehkä tämä on ns. makuasia, mutta tätä tietoa varmaan me tullaan tarvitsemaan muuallakin ja muutenkin noiden stringien ylläpito tällä tavalla on hankalaa

```
enum Direction {
  Up = "UP",
  Down = "DOWN",
  Left = "LEFT",
  Right = "RIGHT",
}

export type IDirection = {
    [key in Direction]: string;
};
```

Auttaisiko esim. ylläoleva koodi. Silloin joutuu ylläpitämään vain enumiä eikä tämän jälkeen tarvita keyofiakaan

Reply:
Eli olisiko tässä ideana että enumissa olisi `ClerkTranslatorBasicInformation`in kentät listattuna, ja alkuperäisen interfacen tilalla olisi tuo `export type ...`? Se varmaan voisi olla mahdollinen, mutta en tiedä kuinka selvä loppujen lopuksi. Onnistuuko myöskään noiden optional-kenttien listaus tuolla mainitsemallasi tavalla?

Reply:
Joo vaikuttaisi siltä että optional kenttien listaus ei ihan onnistu helposti, se menee liian monimutkaiseksi. Mun puolesta voidaan mennä tällä olemassa olevallakin, jos ei halua enumia vielä rinnalle. Palataan takaisin kun tarvitaan noita kenttien nimiä uudestaan.